### DISCUSSION TITLE
Error! Can't parse json file

### DISCUSSION THREAD

**Discussion by koshimazaki** [vercel/next.js#40730] [Help]
### Verify canary release

- [X] I verified that the issue exists in the latest Next.js canary release

### Provide environment information

Operating System:
      Platform: darwin
      Arch: arm64
      Version: Darwin Kernel Version 21.6.0: Sat Jun 18 17:07:22 PDT 2022; root:xnu-8020.140.41~1/RELEASE_ARM64_T6000
    Binaries:
      Node: 16.13.2
      npm: 8.15.1
      Yarn: 3.2.1
      pnpm: N/A
    Relevant packages:
      next: 12.2.4
      eslint-config-next: N/A
      react: 17.0.2
      react-dom: 17.0.2

### How are you deploying your application? (if relevant)

Vercel

### Describe the Bug

I keep getting error once trying to build project with Vercel. It builds fine locally. Is there something i am overlooking? Only json file I have in repo is package.json

Running "vercel build"
--
15:22:31.883 | Vercel CLI 28.0.1
15:22:31.976 | Error! Can't parse json file

### Expected Behavior

Congratulations built with success! ;) 

### Link to reproduction

see json file contents below

### To Reproduce

{
  "private": true,
  "scripts": {
    "dev": "next dev",
    "build": "next build",
    "start": "next start"
  },
  "dependencies": {
    "axios": "^0.27.2",
    "express": "^4.18.1",
    "googleapis": "^105.0.0",
    "next": "latest",
    "react": "^17.0.2",
    "react-dom": "^17.0.2",
    "react-player": "^2.10.1",
    "redux": "^4.2.0",
    "semantic-ui-css": "^2.4.1",
    "semantic-ui-react": "^2.1.3"
  }
}

**Comment by koshimazaki** [comment]
Case sensitive imports / file names was an issue at then end. Exactly as in here: 
https://github.com/vercel/next.js/discussions/18176

Installing ESlint helped and so did not entering the env variables.

**Comment by koshimazaki** [comment]
Just leaving it here in case someone is experiencing the same error. This is related to env variables. It doesn't work for me by inputing Google private key in the Vercel interface. Looks like im not alone. 

https://github.com/vercel/vercel/issues/749

**Comment by wariomx** [comment]
Did you solve it? i hava the same issue, in vercel
Error: Can't parse json file

**Reply by koshimazaki** [reply]
I think at the end it was combination of things as mentioned above. But main one was copying private key without " " like that
 -----BEGIN PRIVATE KEY----- ..... ----- 
 Those two threads mentioned above were really helpful too.

**Comment by ijjk** [comment]
Hi, can you share a repo/deployment where you are seeing this?

**Reply by koshimazaki** [reply]
This is the repo below. My issue was imports with/without capital letters in component names, as I am on a Mac, worked fine locally but not on vercel site. On top of that it was env.variables not working in vercel web interface. Once I copied content within "" from google private key it started working fine.

https://github.com/koshimazaki/ynoetsu
